src/config.c

```diff
@@ -489,11 +489,8 @@ void loadServerConfigFromString(char *config) {
             if ((!strcasecmp(argv[0],config->name) ||
                 (config->alias && !strcasecmp(argv[0],config->alias))))
             {
-                if (argc != 2) {
-                    err = "wrong number of arguments";
-                    goto loaderr;
-                }
```

Comment:
Should it be kept, and changed to `if (argc < 2)`?

Reply:
The `argc` check is now part of the individual config set functions. So it think it was removed by @bjosv because it's now redundant. We know the value is larger than 0 because of the check above (line 480) which should be enough.

Reply:
If "argc" is 1, config such as "save" will be legal, in the old code it would have failed to start server.

Reply:
If "argc" is 1, config such as "save" will be legal, in the old code it would have failed to start server.

Reply:
With argc of 1, the config will be this:

```
# redis.conf
save
```

Reply:
Ohh. Sorry, It seems I was wrong, and the `save` configuration was legal before this.

Reply:
@sundb can you please re-review might last commits.

Reply:
I'm not sure I want to fix this. Before we enforced at least 1 argument to the bind option, today I allow zero arguments but fail because zero arguments means not binding to anything. I think this is ok. Specifically because passing zero arguments via `config set bind ""` is legal (meaning we want to cancel all TCP bindings).
Assuming we have the following conf:
```
# redis.conf
bind ""
unixsocket /tmp/redis.sock
```
I have no problem for it to be treated the same as
```
# redis.conf
bind
unixsocket /tmp/redis.sock
```

Reply:
> I think this is ok. Specifically because passing zero arguments via `config set bind ""` is legal

i think we can look at this differently, it's not the bind handling that's being violated, it's the config file format or config command itself.
i.e. `config set bind` (with argc==3) is an invalid command. and a config file line that just says 'bind' could be invalid too.
it was invalid till now, and although we can make it valid now, it is something we'll never be able to take back (backwards compatibility), so being "conservative" i think i'd rather keep blocking it.

Reply:
Turn out that if we don't allow zero arg configs (like what I did with `bind`) then we end up with a problem because the test suite (and possibly users) creates zero arg `save` config files. For example here:
https://github.com/redis/redis/blob/f1f3cceb50ac97187f9dca37b041583a5f6ee6aa/tests/integration/corrupt-dump-fuzzer.tcl#L96
This creates a `save` with no arguments in the config file. The old code tolerated this because of the non-generic handling of `save` config lines. Now that there's a generic handling we need to decide if we want to support zero args or not **for all** (MLUTI ARG) config lines.
After discussion with @oranagra decide to leave the code as is (i.e. support zero args for bind).